### ISSUE TITLE
Sometimes amount text boxes in the split editor fail getting the focus

### ISSUE THREAD

**Issue by rivaldi8** [codinguser/gnucash-android#417]
Sometimes, when trying to switch the focus from a split memo text box to a amount text box, the focus remains in the memo one. However, the system keyboard is closed and the calculator one is opened. If something is typed with the calculator, the application crashes:
```
E/InputEventReceiver: Exception dispatching input event.
E/MessageQueue-JNI: Exception in MessageQueue callback: handleReceiveCallback
E/MessageQueue-JNI: java.lang.ClassCastException: android.support.v7.widget.AppCompatEditText cannot be cast to org.gnucash.android.ui.util.widget.CalculatorEditText
E/MessageQueue-JNI:     at org.gnucash.android.ui.util.widget.CalculatorKeyboard$1.onKey(CalculatorKeyboard.java:78)
E/MessageQueue-JNI:     at android.inputmethodservice.KeyboardView.detectAndSendKey(KeyboardView.java:827)
E/MessageQueue-JNI:     at android.inputmethodservice.KeyboardView.onModifiedTouchEvent(KeyboardView.java:1349)
E/MessageQueue-JNI:     at android.inputmethodservice.KeyboardView.onTouchEvent(KeyboardView.java:1212)
E/MessageQueue-JNI:     at android.view.View.dispatchTouchEvent(View.java:8486)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2541)
E/MessageQueue-JNI:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2234)
E/MessageQueue-JNI:     at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2365)
E/MessageQueue-JNI:     at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1726)
E/MessageQueue-JNI:     at android.app.Activity.dispatchTouchEvent(Activity.java:2772)
E/MessageQueue-JNI:     at android.support.v7.internal.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:59)
E/MessageQueue-JNI:     at android.support.v7.internal.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:59)
E/MessageQueue-JNI:     at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2326)
E/MessageQueue-JNI:     at android.view.View.dispatchPointerEvent(View.java:8687)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4485)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4343)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3884)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3937)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3903)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4013)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3911)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4070)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3884)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3937)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3903)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3911)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3884)
E/MessageQueue-JNI:     at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6300)
E/MessageQueue-JNI:     at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6256)
E/MessageQueue-JNI:     at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6209)
E/MessageQueue-JNI:     at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6462)
E/MessageQueue-JNI:     at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
E/MessageQueue-JNI:     at android.os.MessageQueue.nativePollOnce(Native Method)
E/MessageQueue-JNI:     at android.os.MessageQueue.next(MessageQueue.java:143)
E/MessageQueue-JNI:     at android.os.Looper.loop(Looper.java:146)
E/MessageQueue-JNI:     at android.app.ActivityThread.main(ActivityThread.java:5532)
E/MessageQueue-JNI:     at java.lang.reflect.Method.invoke(Native Method)
E/MessageQueue-JNI: 	at java.lang.reflect.Method.in
```

To reproduce, just keep tapping between the amount and memo text boxes until it happens.

**Comment by rivaldi8** [comment]
There are two strange things about this issue. One is that it fails in the split editor but not in the transaction editor, although I haven't found anything special about the split editor. The other is that the fix doesn't make much sense.  So, in case the issue appears again, I'll add some notes here with what I've found while looking for the cause.

By logging calls to event handlers I found that when it fails it gets these events:
- 8 touch
- 0 click
- 0 focus

However, when it works it gets these instead:
- 4 touch
- 1 click
- 1 focus

It seems it only fails when the system keyboard is visible.

My bq Aquaris E4 (Android 5, API 21) was affected, but not the Huawei Ascend G300 (Android 2.3.6, API 10).
